vector/src/main/java/im/vector/app/features/roomdirectory/createroom/CreateSubSpaceController.kt

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (c) 2021 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.features.roomdirectory.createroom
+
+import com.airbnb.epoxy.TypedEpoxyController
+import com.airbnb.mvrx.Fail
+import com.airbnb.mvrx.Loading
+import im.vector.app.R
+import im.vector.app.core.epoxy.profiles.buildProfileAction
+import im.vector.app.core.resources.StringProvider
+import im.vector.app.core.ui.list.genericPillItem
+import im.vector.app.features.discovery.settingsSectionTitleItem
+import im.vector.app.features.form.formEditTextItem
+import im.vector.app.features.form.formEditableSquareAvatarItem
+import im.vector.app.features.form.formMultiLineEditTextItem
+import im.vector.app.features.form.formSubmitButtonItem
+import org.matrix.android.sdk.api.session.room.failure.CreateRoomFailure
+import org.matrix.android.sdk.api.session.room.model.RoomJoinRules
+import javax.inject.Inject
+
+class CreateSubSpaceController @Inject constructor(
+        private val stringProvider: StringProvider,
+        private val roomAliasErrorFormatter: RoomAliasErrorFormatter
+) : TypedEpoxyController<CreateRoomViewState>() {
+
+    var listener: CreateRoomController.Listener? = null
+
+    override fun buildModels(viewState: CreateRoomViewState) {
+        // display the form
+        buildForm(viewState, viewState.asyncCreateRoomRequest !is Loading)
+    }
+
+    private fun buildForm(data: CreateRoomViewState, enableFormElement: Boolean) {
+        val host = this
+
+        if (data.isSubSpace) {
```

Comment:
Since we have 2 controllers, and this one is dedicated to creating sub space, this is always true, no?

Reply:
yes done